pkg/controllers/node/termination/suite_test.go

```diff
@@ -452,16 +458,16 @@ var _ = Describe("Termination", func() {
 				}
 				ExpectRequeued(ExpectObjectReconciled(ctx, env.Client, terminationController, node))
 				ExpectNodeWithNodeClaimDraining(env.Client, node.Name)
-				for range podGroup {
-					ExpectSingletonReconciled(ctx, queue)
+				for _, pod := range podGroup {
+					ExpectObjectReconciled(ctx, env.Client, queue, pod)
 				}
 				// Start draining the pod group, but don't complete it yet
 				EventuallyExpectTerminating(ctx, env.Client, lo.Map(podGroup, func(p *corev1.Pod, _ int) client.Object { return p })...)
 
 				// Look at the next pod group and ensure that none of the pods have started terminating on it
 				if i != len(podGroups)-1 {
-					for range podGroups[i+1] {
-						ExpectSingletonReconciled(ctx, queue)
+					for _, pod := range podGroups[i+1] {
+						Expect(queue.Has(pod)).To(BeFalse())
```

Comment:
Is this reconciling the pod?

Reply:
In that case, it seems like the check on L472 is basically not doing anything anymore -- since it takes us reconciling the pod to even get into a state of pods being terminated, I think, like you are saying, we can just check if the pod exists in the queue and (if it doesn't), then we pass the testcase